astropy/time/formats.py

```diff
@@ -370,15 +366,22 @@ def set_jds(self, val1, val2):
         self.jd1 = tm._time.jd1
         self.jd2 = tm._time.jd2
 
-    @property
-    def value(self):
-        # when we get here, getattr will already have ensured our scale
-        # equals epoch scale, so we can just subtract the epoch and convert
-        time_from_epoch = ((self.jd1 - self.epoch.jd1) +
-                           (self.jd2 - self.epoch.jd2)) / self.unit
+    def value(self, parent=None):
+        # Make sure that scale is the same as epoch scale so we can just
+        # subtract the epoch and convert
+        if self.scale != self.epoch_scale:
+            tm = parent.replicate()
```

Comment:
This is going to break if parent is None though.  Meanwhile, since most of the other TimeFormat.value implementations don't even use the `parent` argument it seems like a lot of conceptual overhead added to support one special case.  I suspect there's maybe some other way to work around this--even something like what @mhvk suggested of adding a method on the formatters that can be used internally by `Time.value` to do this (which would still just return `self.value` in most cases).  But keep it an internal method.